src/test/java/org/broadinstitute/hellbender/utils/recalibration/covariates/ContextCovariateUnitTest.java

```diff
@@ -46,12 +50,54 @@ public void testSimpleContexts() {
         }
     }
 
-    public static void verifyCovariateArray(int[][] values, int contextSize, GATKRead read, Covariate contextCovariate, final byte lowQualTail) {
+
+    private static void verifyCovariateArray(final int[][] values, final int contextSize, final GATKRead read, final Covariate contextCovariate, final byte lowQualTail) {
         for (int i = 0; i < values.length; i++) {
             Assert.assertEquals(contextCovariate.formatKey(values[i][0]), expectedContext(read, i, contextSize, lowQualTail), "offset " + i);
         }
     }
 
+    @DataProvider
+    Iterator<Object[]> AnnoyingReads() {
+        final String IUPAC_bases = "NBTRML";
+        final int readLength = 1000;
+        final double pStop = 0.02;
+        final int nTests = 100;
+        final List<Object[]> tests = new ArrayList<>();
+
+        final Random randomGenerator = Utils.getRandomGenerator();
+        for (int i = 0; i < nTests; i++) {
+            final List<Byte> bases = new ArrayList<>();
+            final List<Integer> positions = new ArrayList<>();
+            while (randomGenerator.nextDouble() > pStop) {
+                bases.add(IUPAC_bases.getBytes()[randomGenerator.nextInt(IUPAC_bases.length())]);
+                positions.add(randomGenerator.nextInt(readLength));
+            }
+            tests.add(new Object[]{bases, positions, randomGenerator.nextBoolean(), readLength});
+        }
```

Comment:
I don't think the randomness really adds much value in this DataProvider. Since the individual IUPAC bases are not treated differently by the code, scrambling their relative order doesn't make much difference once you've proved that we can handle each one -- and despite having 100 random cases I think we're actually still missing important test cases that are worth covering.

Recommend rewriting this DataProvider to eliminate the randomness completely and substituting just the following curated list of test cases instead:

* A string of all IUPAC bases (positive strand)
* A string of all IUPAC bases (negative strand)
* A string that starts with an ACGT base, then has a mix of ACGT and IUPAC bases (positive strand)
* A string that starts with an ACGT base, then has a mix of ACGT and IUPAC bases (negative strand)
* A string that starts with an IUPAC base, then has a mix of ACGT and IUPAC bases (positive strand)
* A string that starts with an IUPAC base, then has a mix of ACGT and IUPAC bases (negative strand)
* A string that ends with an ACGT base, and has a mix of ACGT and IUPAC bases (positive strand)
* A string that ends with an ACGT base, and has a mix of ACGT and IUPAC bases (negative strand)
* A string that ends with an IUPAC base, and has a mix of ACGT and IUPAC bases (positive strand)
* A string that ends with an IUPAC base, and has a mix of ACGT and IUPAC bases (negative strand)